### ISSUE TITLE
`messageParams` wrong return type in TS

### ISSUE THREAD

**Issue by lukasaric** [vuelidate/vuelidate#1180]
### Make sure that you are familiar with documentation before submitting an issue.

  https://vuelidate-next.netlify.com/

**Describe the bug**
- When trying to use `messageParams` within the `createI18nMessage` I get the TS error. The reasoning behind it is that it has wrong return type. 

Code example:
```js
 const withI18nMessage = createI18nMessage({
    t: i18n.t.bind(i18n),
    messagePath: ({ $validator }) => `validation.${snakeCase($validator)}`,
    // @ts-ignore
    messageParams: ({ field, max, otherName, property, ...params }) => ({
      ...params,
      field: i18n.t(`form_fields.${snakeCase(field || property)}`),
      max,
      otherField: otherName,
    }),
  })
  
```
The code above is working and the `messageParam`s should basically return factory the same as `params` argument which is:
```js
export function messageParamsFactory(params: {
  model: unknown,
  property: string,
  invalid: boolean,
  pending: boolean,
  propertyPath: string,
  response: unknown,
  validator: string,
  [key: string]: any
}): string;
```
For some reason, currently it says that it needs to return the `string` which does not work

**Expected behavior**

Basically the code above should be changed something like this:
```
type MessageParams = {
  model: unknown,
  property: string,
  invalid: boolean,
  pending: boolean,
  propertyPath: string,
  response: unknown,
  validator: string,
  [key: string]: any
}
export function messageParamsFactory(params: MessageParams): Partial<MessageParams>;
```

**Comment by vaidd4** [comment]
@lukasaric I made a PR for this : https://github.com/vuelidate/vuelidate/pull/1175
I use the code in this PR in my project with [patch-package](https://www.npmjs.com/package/patch-package)
Hope it helps.

**Comment by lukasaric** [comment]
> @lukasaric I made a PR for this : #1175
> I use the code in this PR in my project with [patch-package](https://www.npmjs.com/package/patch-package)
> Hope it helps.

Nice!
